crates/bevy_color/src/srgba.rs

```diff
@@ -331,6 +334,24 @@ const fn decode_hex<const N: usize>(mut bytes: [u8; N]) -> Result<[u8; N], HexCo
     Ok(bytes)
 }
 
+/// Error returned when hex color decoding fails.
+#[derive(Debug, PartialEq, Eq)]
+pub enum HexColorError {
+    /// Unexpected length of hex string
+    Length,
+    /// Invalid hex character
+    Char(char),
+}
+
+impl core::fmt::Display for HexColorError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Length => write!(f, "Unexpected length of hex string"),
+            Self::Char(char) => write!(f, "Invalid hex char {}", char),
+        }
+    }
+}
+
```

Comment:
IMO we should continue using `thiserror`. It is already a common dependency on bevy and make error handling nicer and easier.

Reply:
I do agree in principle, `thiserror` is my go-to choice for error handling in Rust. My only reason for not including it here is that `bevy_color` currently has a single fallible function, which only needs a small `Display` implementation. I would rather wait until a second fallible function is added before investing in the extra dependency personally. But this isn't a strong opinion of mine, I'm happy to yield if anyone else agrees with you.

Reply:
Fair enough, in the end it will depend if this PR is merged before or after #12082 where I have a use case for `thiserror`.